server/routes.go

```diff
@@ -666,61 +667,13 @@ func (s *Server) EmbedHandler(c *gin.Context) {
 		return
 	}
 
-	kvData, _, err := getModelData(m.ModelPath, false)
-	if err != nil {
-		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
-		return
-	}
-
-	var count int
-	for i, s := range input {
-		tokens, err := r.Tokenize(c.Request.Context(), s)
-		if err != nil {
-			c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
-			return
-		}
-
-		ctxLen := min(opts.NumCtx, int(kvData.ContextLength()))
-		if len(tokens) > ctxLen {
-			if !truncate {
-				c.JSON(http.StatusBadRequest, gin.H{"error": "input exceeds maximum context length"})
```

Comment:
There's a slight difference in behavior between the old and new code. Previously, if we exceed the context length without truncate we would return `http.StatusBadRequest`, now we will return `http.StatusInternalServerError`. This is both a little bit of an API change, as people could be checking the error code and BadRequest is more accurate. It's probably best if we can maintain the existing behavior.